pkg/crc/cluster/cluster.go

```diff
@@ -119,20 +120,82 @@ func UpdateClusterID(ocConfig oc.Config) error {
 	return nil
 }
 
-func AddProxyConfigToCluster(ocConfig oc.Config, proxy *network.ProxyConfig) error {
-	cmdArgs := []string{"patch", "proxy", "cluster", "-p",
-		fmt.Sprintf(`'{"spec":{"httpProxy":"%s", "httpsProxy":"%s", "noProxy":"%s"}}'`, proxy.HTTPProxy, proxy.HTTPSProxy, proxy.GetNoProxyString()),
-		"-n", "openshift-config", "--type", "merge"}
+func AddProxyConfigToCluster(sshRunner *ssh.Runner, ocConfig oc.Config, proxy *network.ProxyConfig) error {
+	type trustedCA struct {
+		Name string `json:"name"`
+	}
+
+	type proxyPatch struct {
+		HTTPProxy  string    `json:"httpProxy"`
+		HTTPSProxy string    `json:"httpsProxy"`
+		NoProxy    string    `json:"noProxy"`
+		TrustedCA  trustedCA `json:"trustedCA"`
+	}
+
+	type spec struct {
+		Spec proxyPatch `json:"spec"`
+	}
+
+	patch := &spec{
+		Spec: proxyPatch{
+			HTTPProxy:  proxy.HTTPProxy,
+			HTTPSProxy: proxy.HTTPSProxy,
+			NoProxy:    proxy.GetNoProxyString(),
+		},
+	}
 
 	if err := WaitForOpenshiftResource(ocConfig, "proxy"); err != nil {
 		return err
 	}
+
+	if proxy.ProxyCACert != "" {
+		trustedCAName := "user-ca-bundle"
+		logging.Debug("Adding proxy CA cert to cluster")
+		if err := addProxyCACertToCluster(sshRunner, ocConfig, proxy, trustedCAName); err != nil {
+			return err
+		}
+		patch.Spec.TrustedCA = trustedCA{Name: trustedCAName}
+	}
+
+	patchEncode, err := json.Marshal(patch)
+	if err != nil {
+		return fmt.Errorf("Failed to encode to json: %v", err)
+	}
+	logging.Debugf("Patch string %s", string(patchEncode))
+
+	cmdArgs := []string{"patch", "proxy", "cluster", "-p", fmt.Sprintf("'%s'", string(patchEncode)), "-n", "openshift-config", "--type", "merge"}
```

Comment:
Is the `fmt.Sprintf` required?

Reply:
@cfergeau yes because we are executing command as ssh runner so it should be passed as `'<json_string>'` otherwise you will get error about json patch not valid.

Reply:
oh I did not noticed the added quotes ^^